pkg/location/location_test.go

```diff
@@ -107,18 +112,95 @@ func (s *LocationSuite) TearDownTest(c *C) {
 		err := s.root.Delete(ctx, s.testpath)
 		if err != nil {
 			c.Log("Cannot cleanup test directory: ", s.testpath)
-			return
+		}
+	}
+	if s.testMultipartPath != "" {
+		c.Assert(s.root, NotNil)
+		ctx := context.Background()
+		err := s.root.Delete(ctx, s.testMultipartPath)
+		if err != nil {
+			c.Log("Cannot cleanup test directory: ", s.testMultipartPath)
 		}
 	}
 }
 
 func (s *LocationSuite) TestWriteAndReadData(c *C) {
 	ctx := context.Background()
-	teststring := "test-content"
+	teststring := "test-content-check"
 	err := writeData(ctx, s.osType, s.profile, bytes.NewBufferString(teststring), s.testpath)
 	c.Check(err, IsNil)
 	buf := bytes.NewBuffer(nil)
 	err = readData(ctx, s.osType, s.profile, buf, s.testpath)
 	c.Check(err, IsNil)
 	c.Check(buf.String(), Equals, teststring)
 }
+
+func (s *LocationSuite) TestAzMultipartUpload(c *C) {
+	if s.osType != objectstore.ProviderTypeAzure {
+		c.Skip(fmt.Sprintf("Not applicable for location type %s", s.osType))
+	}
+
+	// Create dir if not exists
+	_, err := os.Stat(s.suiteDirPrefix)
+	if os.IsNotExist(err) {
+		err := os.MkdirAll(s.suiteDirPrefix, 0755)
+		c.Check(err, IsNil)
+	}
+	// Create test file
+	f, err := os.Create(s.testMultipartPath)
+	c.Check(err, IsNil)
+	defer f.Close()
+
+	ctx := context.Background()
+	for _, fileSize := range []int64{
+		0,                 // empty file
+		100 * 1024 * 1024, // 100M ie < buffSize
+		300 * 1024 * 1024, // 300M ie > buffSize
+	} {
+		// Create dump file
+		err := os.Truncate(s.testMultipartPath, fileSize)
+		c.Assert(err, IsNil)
+		testcontent, err := ioutil.ReadFile(s.testMultipartPath)
```

Comment:
`os.Open()` will give you a reader - you don't need to get the content in-memory then create a bytes.Buffer.

Reply:
We are re-using file pointer created at https://github.com/kanisterio/kanister/pull/809/files/147b928221ac56008f56f83ce76f2e525c764eee#diff-c1b47cb14ad134412666df8eff6311f52edc385a3e6ffa6a15564f24ec9b297bR150